src/index.js

```diff
@@ -123,18 +123,14 @@ class UglifyJsPlugin {
               let input;
               let inputSourceMap;
 
-              if (this.options.sourceMap) {
-                if (asset.sourceAndMap) {
-                  const sourceAndMap = asset.sourceAndMap();
-                  inputSourceMap = sourceAndMap.map;
-                  input = sourceAndMap.source;
-                } else {
-                  inputSourceMap = asset.map();
-                  input = asset.source();
-                }
+              if (this.options.sourceMap && asset.sourceAndMap) {
+                const sourceAndMap = asset.sourceAndMap();
```

Comment:
```js
const { source, map } = asset.sourceAndMap();

input = source;
inputSourceMap = map;

sourceMap = new SourceMapConsumer(inputSourceMap)
```

Reply:
Does `webpack >= v2.0.0` already support this convenience method ? So the `else` branch  **is** really legacy

Reply:
@michael-ciniawsky need investigate

Reply:
Should be safe https://github.com/webpack/webpack/blob/webpack-2/package.json#L26